indexing-service/src/main/java/org/apache/druid/indexing/common/task/Task.java

```diff
@@ -160,8 +160,11 @@ default int getPriority()
   /**
    * Execute preflight actions for a task. This can be used to acquire locks, check preconditions, and so on. The
    * actions must be idempotent, since this method may be executed multiple times. This typically runs on the
-   * coordinator. If this method throws an exception, the task should be considered a failure.
-   * <p/>
+   * Overlord. If this method throws an exception, the task should be considered a failure.
+   *
+   * This method will not necessarily be executed before {@link #run(TaskToolbox)}, since this task readiness check
+   * may be done on a different machine from the one that actually runs the task.
```

Comment:
Hmm, how can this happen? I thought `Task.isReady()` is always called before `Task.run()` because a task can be scheduled only when `Task.isReady()` returns true. `Task.run()` will be called after the task is scheduled in some indexer or middleManager.

Reply:
It won't necessarily be called on that same Task object (it might be called on a different instance that represents the same actual task).

Reply:
Oh, I see. I think it happens only in indexers because peon calls `isReady()` before it runs its task.